plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -53,7 +53,7 @@ func (l *lifecycle) Admit(a admission.Attributes) (err error) {
 
 	// prevent deletion of immortal namespaces
 	if a.GetOperation() == admission.Delete && a.GetKind() == "Namespace" && l.immortalNamespaces.Has(a.GetName()) {
-		return errors.NewForbidden(a.GetKind(), a.GetName(), fmt.Errorf("namespace can never be deleted"))
+		return errors.NewForbidden(a.GetKind(), a.GetName(), fmt.Errorf("an immortal namespace can never be deleted"))
```

Comment:
"may not" implies policy, and that's exactly right, so I think that works.

On Tue, Oct 20, 2015 at 2:35 PM, Derek Carr <notifications@github.com>
wrote:

> In plugin/pkg/admission/namespace/lifecycle/admission.go
> <https://github.com/kubernetes/kubernetes/pull/15942#discussion_r42557806>
> :
>
> > @@ -53,7 +53,7 @@ func (l *lifecycle) Admit(a admission.Attributes) (err error) {
> >
> >  	// prevent deletion of immortal namespaces
> >  	if a.GetOperation() == admission.Delete && a.GetKind() == "Namespace" && l.immortalNamespaces.Has(a.GetName()) {
> > -		return errors.NewForbidden(a.GetKind(), a.GetName(), fmt.Errorf("namespace can never be deleted"))
> > +		return errors.NewForbidden(a.GetKind(), a.GetName(), fmt.Errorf("an immortal namespace can never be deleted"))
>
> @thockin <https://github.com/thockin> - i do not want to expose the
> concept of immortal to an end-user.
>
> I am good with "namespace may not be deleted"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15942/files#r42557806>.
>